File: GUISM_SERVER/source/raca/Raca.java
package raca;

import descendencia.Descendencia;
import especializacao.Especializacao;
import habilidade.Habilidade;
import habito.Habito;
import idioma.Idioma;
import kikaha.jdbi.serializers.Column;
import kikaha.jdbi.serializers.Entity;
import lombok.Data;
import patrono.Patrono;

import java.util.Set;

@Data
@Entity
public class Raca{

    @Column
    Long idRaca;

    @Column
    String nomeRaca;

    @Column
    String raridadeRaca;

    @Column
    String descRaca;

    @Column
    String longevidadeRaca;

    @Column
    String TRACOSFISIOLOGICOS;

    @Column
    String culturaRaca;

    @Column
    String historiaRaca;

    @Column
    Long racaForca;

    @Column
    Long racaConstituicao;

    @Column
    Long racaAgilidade;

    @Column
    Long racaDestreza;

    @Column
    Long racaInteligencia;

    @Column
    Long racaSabedoria;

    @Column
    Long racaCarisma;

    @Column
    Long sangue;

    @Column
    Long vigor;

    Set<Descendencia> descendencias;

    Set<Especializacao> especializacoes;

    Set<Habilidade> habilidades;

    Set<Habito> habitos;

    Set<Idioma> idiomas;

    Set<Patrono> patronos;
}

File: src/components/raca/Raca.ts
import Idioma from "../idioma/Idioma";
import Patrono from "../patrono/Patrono";
import Descendencia from "../descendencia/Descendencia";
import Especializacao from "../especializacao/Especializacao";
import Habito from "../habito/Habito";
import Habilidade from "../habilidade/Habilidade";

export default interface Raca {
    idRaca?: number;
    nomeRaca?: string;
    raridadeRaca?: string;
    descRaca?: string;
    longevidadeRaca?: string;
    culturaRaca?: string;
    historiaRaca?: string;
    racaForca?: number;
    racaConstituicao?: number;
    racaAgilidade?: number;
    racaDestreza?: number;
    racaInteligencia?: number;
    racaSabedoria?: number;
    racaCarisma?: number;
    sangue?: number;
    vigor?: number;
    descendencias?: Descendencia[];
    especializacoes?: Especializacao[];
    habilidades?: Habilidade[];
    habitos?: Habito[];
    idiomas?: Idioma[];
    patronos?: Patrono[];
    tracosfisiologicos?: string;
}
